### ISSUE TITLE
[unbound-method] Add "ignoreStatic" to "unbound-method" in recommendations

### ISSUE THREAD

**Issue by jacob87o2** [typescript-eslint/typescript-eslint#1639]
The `@typescript-eslint/recommended-requiring-type-checking` set of rules contain `unbound-method` with default options. 

However, there is no sense to report unbound method error, if the method is static. Therefore, we can use handy option - `ignoreStatic`, like so: 

```JSON
"@typescript-eslint/unbound-method": [
  "error",
  {
    "ignoreStatic": true
  }
],
```

I made a PR #1636, but was asked to create an issue instead, so here it is.

**Comment by bradzacher** [comment]
Static members can reference the `this` context. It is pretty weird, but in a static context, `this` references the class itself:

```ts
class Foo {
    static log() {
        console.log(this.name);
    }
    static bar() {
        this.log();
    }
}
```
[repl](https://www.typescriptlang.org/play/?ssl=1&ssc=1&pln=11&pc=1#code/MYGwhgzhAEBiD29oG8BQ0PQgFzNglsNCPAOYAUAlCupncPAHYTwgCmAdCRdgBb4QOjMAFs2lANy0MAX2lZcBIgCMwAJyo06dPgK5kqUunLmpUQA)

![image](https://user-images.githubusercontent.com/7462525/75277137-50d7a780-57bc-11ea-8b34-d8c9605d8950.png)

So the question is:
Is this something users actively do?
Is it something this rule should catch by default, or should it be opt-in because it's a weird case, and the majority case is to not use `this` in a static context?

**Comment by jacob87o2** [comment]
Technically, you're right. I myself don't use `this` in static methods, because it does feel weird (would be nice to have separate keyword to do that). But somebody might have different opinion. 

Here is a little more context on my issue with the recommended rule. I had class with static method:
```ts
class SomeClass {
  static someStaticFunction(...) { ... }
}
```
And I used it in array map, like this:
```ts
someArray.map(SomeClass.someStaticFunction);
```
It was reported as unbound method.

One way to solve this is to add `unbound-method` rule, with `useStatic` option in eslintrc file.

But another workaround I found was to convert it to array function.
```ts
class SomeClass {
  static someStaticFunction = (...) => { ... }
}
```
This not only fixes the unbound method error (I guess because of arrow function _lexical context_) but also disallows the quirky use of `this` in static method: 
> 'this' cannot be referenced in a static property initializer

So, if nobody else will comment on the issue - I guess we can close it.
